Can't set the value of a DateTimeField with auto_now or auto_now_add #23

rodbv opened the pull request:
**Describe the change**
A clear and concise description of what the change is.

**PR Checklist**
- [x] Change is covered with tests
- [x] [CHANGELOG.md](CHANGELOG.md) is updated

rodbv commented:
I've faced an issue similar to [this](https://github.com/model-bakers/model_bakery/issues/23#issuecomment-800589250) and gave it a go. One issue with this implementation is that it only works if either `_refresh_after_create` is set to `True`, or the user calls `refresh_from_db()` directly like I did on the test.

I could call `instance.refresh_from_db()` [here](https://github.com/model-bakers/model_bakery/pull/451/files#diff-e5857deb915e241f429a0c118e89e06a3388d3ce1466e3aa4b960b7055172b6dR625) but this would override `_refresh_after_create`

amureki commented:
@rodbv oh, many thanks for this attempt to tackle an annoying issue! 👍 This is promising!
I would need to dig more into the implementation, but meanwhile, may I ask you to look into the failing tests? I think, there are edge cases here that are causing errors.

Best,
Rust

rodbv commented:
@amureki thanks! I've fixed the tests and the linter error. When running the tests workflow on my repo there's an error on the ruff linter but it seems to be a setup issue.

rodbv commented:
I've dealt with my test failing depending on the use_tz setting. Ruff linter is still failing, is there anything I can do?

amureki commented:
> I've dealt with my test failing depending on the use_tz setting. Ruff linter is still failing, is there anything I can do?

Thanks for fixing the tests! I'll take a look into tz stuff before the end of this week. And also will do a proper check on your code.
Ruff part should be resolved if you update your branch on the main.

Sorry for the complications on the contribution way.

amureki commented:
@rodbv I finally got some time and figured the issue. This was happening due to `django.conf.settings` tampering, which is usually not a good practice (but I know, we still have it in a couple of tests).
I updated your test to use [a fixture provided by pytest-django](https://pytest-django.readthedocs.io/en/latest/helpers.html#settings), which solved the issue.

I also went ahead and fixed the last ruff problem, so it is good to go now, I am going to merge it and release now!

amureki commented:
I released it as [1.17.0](https://github.com/model-bakers/model_bakery/releases/tag/1.17.0)

rodbv commented:
> @rodbv I finally got some time and figured the issue. This was happening due to `django.conf.settings` tampering, which is usually not a good practice (but I know, we still have it in a couple of tests). I updated your test to use [a fixture provided by pytest-django](https://pytest-django.readthedocs.io/en/latest/helpers.html#settings), which solved the issue.
> 
> I also went ahead and fixed the last ruff problem, so it is good to go now, I am going to merge it and release now!

Awesome! Thanks a lot for the support and for being so welcoming, I'm looking forward to contribute more, I'll have another look at the list of open issues :)

amureki commented:
@rodbv well, thank you for the contribution, patience and kind words, much appreciated! ✨
